project/5000user5000/README.rst

```diff
@@ -0,0 +1,154 @@
+LUT-based Mixed-Precision GEMM
+==============================
+
+
+Basic Information
+=================
+
+- **Project**: mpGEMM
+- **Github**: https://github.com/5000user5000/mpGEMM
+
+On resource-constrained devices, such as embedded systems, running deep 
+learning models, especially LLMs, is highly computationally expensive. Low-bit 
+quantization is a popular solution to reduce memory consumption. However, when 
+weights are quantized below 8 bits, performing matrix multiplication between 
+the weight matrix and an FP16 activation matrix requires dequantization to a 
+common precision, as hardware lacks native support for mixed-precision matrix 
+multiplication (mpGEMM). Some recent research suggests using lookup tables 
+(LUTs) to replace dequantization, further reducing computational overhead.
+
+.. image:: ./img/lut.png
+
+
+Problem to Solve
+================
+
+- Support mixed-precision matrix computation (weight INT1~4) where activation
+tensors are limited to FP16.
+
+- Implement **precomputed lookup table (LUT)-based computation** to accelerate
+low-bit matrix multiplications.
+
+- Enable the LUT to reside in the fastest on-chip memory and parallel lookup.
+
+- Preliminary estimates suggest that vendor libraries such as MKL or
+Accelerate can be **10× to 1000×** faster than naive GEMM implementations.
+This project aims to evaluate how closely LUT-based methods can approach these
+performance levels in mixed-precision low-bit GEMM.
+
+Prospective Users
+=================
+
+This project will benefit:
+
+1. **Embedded AI Developers**: Those optimizing AI models for deployment on
+edge devices.
+
+2. **Data Scientists**: Users who need optimized inference on diverse hardware
+setups.
+
+3. **Academia & Research Labs**: Those investigating numerical optimization
+and quantization techniques.
+
+
+System Architecture
+===================
+
+The project consists of the following components:
+
+**Lookup Table Component**
+
+- Precomputes and stores the LUT in the on-chip memory.
+
+- Uses SIMD instructions (intel AVX) to accelerate lookup.
+
+**GEMM Component**
+
+- Implements the  matrix multiplication algorithm in C++.
+
+- With 3 modes: **LUT-based**, **naive**, and **vendor** libraries
+(MKL/Accelerate).
+
+- Supports mixed-precision matrix computation (weight INT1~4, activation FP16).
+
+**Benchmarking Component**
+
+- The benchmarking Component will compare latency between LUT-based GEMM, vendor
+libraries, and traditional dequantization-based methods.
+
+
+
+*****************
+ API Description
+*****************
+
+.. code-block:: python
+
+  import mpGEMM
+
+  # Initialize GEMM engine (Choose backend: "lut", "mkl", "naive")
+  gemm = mpGEMM(backend="lut", use_simd=True)
+
+  # Perform GEMM operation, specifying weight bit-width
+  output = gemm.matmul(weights, activations, weight_bit_width=4)
```

Comment:
Focusing on only matmul will make your Python interface too boring to make presentations.  Keep it in mind and see how it goes.